File: NZSLDict/Classes/DiagramView.m
////
////  DiagramView.m
////  NZSL Dict
////
////  Created by Greg Hewgill on 27/04/13.
////
////
//
//#import "DiagramView.h"
//#import "NZSLDict-Swift.h"
//
//@implementation DiagramView {
//    DetailView *detailView;
//    UIImageView *imageView;
//}
//
//- (id)initWithFrame:(CGRect)frame
//{
//    self = [super initWithFrame:frame];
//    if (self) {
//        // Initialization code
//        self.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
//        
//        detailView = [[DetailView alloc] initWithFrame:CGRectMake(0, 0, self.bounds.size.width, DetailView.height)];
//        detailView.autoresizingMask = UIViewAutoresizingFlexibleWidth;
//        [self addSubview:detailView];
//    
//        imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, DetailView.height, self.bounds.size.width, self.bounds.size.height-DetailView.height)];
//        imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
//        imageView.backgroundColor = [UIColor whiteColor];
//        imageView.contentMode = UIViewContentModeScaleAspectFit;
//        [self addSubview:imageView];
//    }
//    return self;
//}
//
//- (void)showEntry:(DictEntry *)entry
//{
//    [detailView showEntry:entry];
//    imageView.image = [UIImage imageNamed:entry.image];
//}
//
//@end

File: NZSLDict/Classes/DiagramView.swift
import UIKit

class DiagramView: UIView {
    var detailView: DetailView!
    var imageView: UIImageView!

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor(named: "app-background")

        detailView = DetailView()
        detailView.translatesAutoresizingMaskIntoConstraints = false
        self.addSubview(detailView)

        let detailLeading = detailView.leadingAnchor.constraint(equalTo: self.leadingAnchor)
        let detailTrailing = detailView.trailingAnchor.constraint(equalTo: self.trailingAnchor)
        let detailTop = detailView.topAnchor.constraint(equalTo: self.topAnchor)
        let detailHeight = detailView.heightAnchor.constraint(equalToConstant: DetailView.height)

        NSLayoutConstraint.activate([
            detailLeading,detailTrailing,detailTop,detailHeight
        ])

        imageView = UIImageView()
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.contentMode = .scaleAspectFit
        self.addSubview(imageView)

        let ivLeading = imageView.leadingAnchor.constraint(equalTo: self.leadingAnchor)
        let ivTrailing = imageView.trailingAnchor.constraint(equalTo: self.trailingAnchor)
        let ivTop = imageView.topAnchor.constraint(equalTo: detailView.bottomAnchor, constant: 10)
        let ivBottom = imageView.bottomAnchor.constraint(equalTo: self.bottomAnchor)

        NSLayoutConstraint.activate([
            ivLeading,ivTrailing,ivTop,ivBottom
        ])

    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func showEntry(_ entry: DictEntry) {
        detailView.showEntry(entry)


        if #available(iOS 13.0, *) {
            imageView.tintColor = UIColor(named: "diagram-tint")
            imageView.image = UIImage(named: entry.image)?.withRenderingMode(.alwaysOriginal)
        } else {
            imageView.image = UIImage(named: entry.image)
        }
    }
}
